File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon/stats/Stats.java
package pt.rfsfernandes.model.pokemon.stats;

import com.google.gson.annotations.SerializedName;

import pt.rfsfernandes.model.SimpleModelData;

/**
 * Class Stats created at 1/16/21 15:31 for the project PoketDex
 * By: rodrigofernandes
 */
public class Stats {
  @SerializedName("base_stat")
  private int baseStat;

  @SerializedName("stat")
  private SimpleModelData stat;

  public Stats(int baseStat, SimpleModelData stat) {
    this.baseStat = baseStat;
    this.stat = stat;
  }

  public int getBaseStat() {
    return baseStat;
  }

  public void setBaseStat(int baseStat) {
    this.baseStat = baseStat;
  }

  public SimpleModelData getStat() {
    return stat;
  }

  public void setStat(SimpleModelData stat) {
    this.stat = stat;
  }

  @Override
  public String toString() {
    return "Stats{" +
        "baseStat=" + baseStat +
        ", stat=" + stat +
        '}';
  }
}

File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon/stats/Stats.kt
package pt.rfsfernandes.model.pokemon.stats

import pt.rfsfernandes.model.SimpleModelData
import com.google.gson.annotations.SerializedName
import androidx.room.TypeConverters
import pt.rfsfernandes.data.local.PokemonTypeConverters

/**
 * Class Stats created at 1/16/21 15:31 for the project PoketDex
 * By: rodrigofernandes
 */
data class Stats(
        @SerializedName("base_stat")
        var baseStat: Int,
        @TypeConverters(PokemonTypeConverters::class)
        @SerializedName("stat")
        var stat: SimpleModelData
) {

    override fun toString(): String {
        return "Stats{" +
                "baseStat=" + baseStat +
                ", stat=" + stat +
                '}'
    }
}